src/qt/transactionrecord.cpp

```diff
@@ -42,7 +42,80 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
     uint256 hash = wtx.GetHash();
     std::map<std::string, std::string> mapValue = wtx.mapValue;
 
-    if (nNet > 0 || wtx.IsCoinBase())
+    bool involvesWatchAddress = false;
```

Comment:
This looks strange?! Do you only want to include `bool involvesWatchAddress = false;` into the if condition? If so, i think a intent would be nice.